packages/edit-site/src/hooks/template-part-edit.js

```diff
@@ -70,12 +68,16 @@ export const withEditBlockControls = createHigherOrderComponent(
 		const { attributes, name } = props;
 		const isDisplayed = name === 'core/template-part' && attributes.slug;
 
+		if ( ! isDisplayed ) {
+			return <BlockEdit { ...props } />;
```

Comment:
Should we use `key` here and below to avoid accidental remounting?

Reply:
When does it remount? Block names cannot change. I guess the slug can change. Maybe we should just remove that from the condition.

Reply:
Mainly to safeguard and the best practice. Remounting can have unexpected side effects - #50292.

> Maybe we should just remove that from the condition.

The editor cannot display the template part in focus mode without slug in focus mode, so we need this condition.

Reply:
In that case I think we need to make sure it remains the same component instance and within the same fragment. A key won't help with that I think? I changed it so it's the same instance for both conditions, and I added the key just to be sure.

Reply:
It does help when the component's position changes after the initial render. If the key is provided, React will use it as a part of the position instead of the order in the parent component.

This example won't retain the state between toggles without the `key` - https://codesandbox.io/p/sandbox/funny-herschel-slys5c?file=%2Fsrc%2FApp.js%3A33%2C29